src/com/google/javascript/jscomp/RewriteDynamicImports.java

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright 2021 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.javascript.jscomp;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.javascript.jscomp.ModuleRenaming.GlobalizedModuleName;
+import com.google.javascript.jscomp.deps.ModuleLoader;
+import com.google.javascript.jscomp.deps.ModuleLoader.ModulePath;
+import com.google.javascript.jscomp.modules.Module;
+import com.google.javascript.jscomp.modules.ModuleMap;
+import com.google.javascript.jscomp.parsing.parser.FeatureSet.Feature;
+import com.google.javascript.rhino.IR;
+import com.google.javascript.rhino.Node;
+import com.google.javascript.rhino.Token;
+import com.google.javascript.rhino.jstype.JSType;
+import com.google.javascript.rhino.jstype.JSTypeNative;
+import com.google.javascript.rhino.jstype.JSTypeRegistry;
+import com.google.javascript.rhino.jstype.TemplatizedType;
+import javax.annotation.Nullable;
+
+/**
+ * Rewrite dynamic import expressions to account for bundling and module rewriting. Since dynamic
+ * imports cannot be fully polyfilled, optionally support replacing the expression with a function
+ * call which indicates an external polyfill is utilized.
+ *
+ * <p>If the import specifier is a string literal and the module resolver recognizes the target, the
+ * pass retargets the specifier to the correct output chunk.
+ *
+ * <p>Example from:
+ *
+ * <pre>
+ *   import('./input-module1.js');
+ * </pre>
+ *
+ * To:
+ *
+ * <pre>
+ *   imprt_('./output-chunk0.js').then(function() { return module$output$chunk0; });
+ * </pre>
+ */
+public class RewriteDynamicImports extends NodeTraversal.AbstractPostOrderCallback
+    implements CompilerPass {
+
+  static final DiagnosticType UNABLE_TO_COMPUTE_RELATIVE_PATH =
+      DiagnosticType.error(
+          "JSC_UNABLE_TO_COMPUTE_RELATIVE_PATH",
+          "Unable to compute relative import path from \"{0}\" to \"{1}\"");
+
+  private final AbstractCompiler compiler;
+  private final AstFactory astFactory;
+  private final String alias;
+  private boolean dynamicImportsRemoved = false;
```

Comment:
It looks like this field isn't really needed anymore.

Our internal automated checks report several warnings at its usages, including saying that the place where its value is used it could be a local variable.

Reply:
I think this one may be a bug in our internal automated checks. I'll file a bug internally

Reply:
It's definitely used.